### PR TITLE
RFC: New container.New() and container.NewWithoutLayout() functions

### PR DESCRIPTION
**Pull request by Jacalz**
### Description:

This is a proposal about adding `container.New()` and `container.NewWithoutLayout()` to the container package, like discussed.

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

#### Where applicable:

- [x] Public APIs match existing style.
- [x] Any breaking changes have a deprecation path or have been discussed.

### CODE REVIEW

**Comment by andydotxyz** [comment]
Thanks for this, seems like a good idea.
However I'm having second thoughts about the `NewWithoutLayout` naming - without the `NewWithLayout` alongside it doesn't seem quite as logical - could a shorter version be found?

**Comment by stuartmscott** [comment]
I thought the same, perhaps `New` and `NewWithLayout` would be better, though it implies the default is to use a container without a layout :/

**Comment by andydotxyz** [comment]
Nah, the default certainly needs to be with-layout.
I guess `NewManual` or something like that would fit the naming in this package?

**Comment by stuartmscott** [comment]
Maybe we're conflating two separate things. I agree container with layout should be the default. Do we need to support container without a layout, what are the use cases?

**Comment by andydotxyz** [comment]
Yes we need to support a container without layout. It's called "Manual layout" and is a way for people to position items in a container individually - whereas a layout would work on the anonymous slice of elements.

**Comment by stuartmscott** [comment]
How is that different from `fyne.NewContainerWithLayout(MyCustomLayout(), items...)`?

**Comment by andydotxyz** [comment]
> How is that different from `fyne.NewContainerWithLayout(MyCustomLayout(), items...)`?

Well, like I thought I said - manual layout operates on individual named elements, a layout operates on an anonymous slice of elements.

Maybe think of `WithoutLayout` like the Swing `nil` layout? It's a useful way to control small areas or drawing elements.

**Comment by Jacalz** [comment]
I think that I would prefer going with `.New()` and `.NewManual()`. It is the best names we have, in my opinion at least.

**Comment by andydotxyz** [comment]
> I think that I would prefer going with `.New()` and `.NewManual()`

I think that's pretty clear and concise, happy to go with that.

**Comment by stuartmscott** [comment]
What value does it provide other than being a struct containing a slice of fyne.CanvasObject?

**Comment by andydotxyz** [comment]
No other value that I can think of. The `fyne.Container` exists to include multiple elements where otherwise only one would be possible.
Usually this contains a layout to help arrange these, but it is not required.

**Comment by Jacalz** [comment]
Updated this PR with new naming.

**Review by stuartmscott — CHANGES_REQUESTED** [review]
I disagree with the 'manual' naming. Manual implies the alternative is Automatic, and as all of this is code nothing happens manually ie there is no human work to lay these objects out.

Perhaps container is conflating two things, one which just groups a set of canvasobjects together, and another which lays out a group of canvasobjects.

**Comment by andydotxyz** [comment]
Hmm, not sure. In a way using a `Layout` is automatic. it adapts to size and reconfigures appropriately.
The "manual" cannot so does require additional work to get a similar outcome...

**Comment by stuartmscott** [comment]
Both constructors provide something that can hold multiple canvasobjects, so that is the core functionality. The ability to provide a layout for arranging said canvasobjects is an optional extra, which brings me back to `New` and `NewWithLayout`

**Comment by andydotxyz** [comment]
We reversed the default from  `NewContainer` and `NewContainerWithLayout` in the `fyne` package to `NewContainer` and `NewContainerWithoutLayout` because it was causing confusion.
People were using the default, shorter, constructor and then complaining that layouts didn't work and that containers were difficult to use.
I don't want to revert to that state. Whilst the layout is optional it should not be default omitted.

**Comment by Jacalz** [comment]
I have one suggestion that might kind of work. Could we not have just one `container.New()` function that takes and the objects. We can then and let manual layout happen with passing a `nil` layout just like we now pass `nil` to icons if we don't want them in for example a `widget.Card` or `widget.Icon`. I know that passing `nil` isn't the most intuitive way, but it is the best solution that I can fink of and it seems sensible in my opinion.

**Comment by andydotxyz** [comment]
It would probably be an acceptable solution - though it doesn't feel great.
If you're deprecating the old methods we would end up with a deprecation for `NewContainerWIthoutLayout` recommending `container.New(nil)` which feels a bit of a backwards step...

**Comment by Jacalz** [comment]
Reverted back to the old naming. It seemed pretty clear on the call today that `container.NewWithoutLayout()` might be the best we have at the moment.

**Review by stuartmscott — APPROVED** [review]

File: container.go [hunk 1]
```diff
@@ -15,14 +15,13 @@ type Container struct {
 }
 
 // NewContainer returns a new Container instance holding the specified CanvasObjects.
-//
```
Comment (andydotxyz) [inline container.go:18]: This bank line is required.
Also please add one to the other deprecated notices - it needs to begin a new paragraph.

**Review by andydotxyz — CHANGES_REQUESTED** [review]

File: container/container.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// Package container provides container widgets that are used to lay out and organise applications
+package container
+
+import (
+	"fyne.io/fyne"
+)
+
+// New returns a new Container instance holding the specified CanvasObjects which will be laid out according to the specified Layout.
```
Comment (stuartmscott) [inline container/container.go:8]: Needs "Since: 2.0.0" annotations

**Review by stuartmscott — CHANGES_REQUESTED** [review]

**Comment by Jacalz** [comment]
Should be fixed now :)

**Review by stuartmscott — APPROVED** [review]

**Review by andydotxyz — APPROVED** [review]

**Pull request merged**